tests/testcases/040_check-network-adv.yml

```diff
@@ -37,7 +37,7 @@
       until: ncs_pod.stdout.find('Running') != -1
       retries: 3
       delay: 10
-      when: inventory_hostname == groups['kube_control_plane'][0]
+      when: inventory_hostname == groups['kube_control_plane'] | first
```

Comment:
Is that necessary to make the PR works ? It does not change the semantic right ?

If not I'd drop those, the PR is already big enough ! ^^